### DISCUSSION TITLE
Should a warning be raised when some routes can never be called ?

### DISCUSSION THREAD

**Discussion by Tyrannas** [fastapi/fastapi#7110] [Questions]
### First check

* [x] I added a very descriptive title to this issue.
* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [x] I already read and followed all the tutorial in the docs and didn't find an answer.
* [x] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [x] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [x] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [x] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```Python
from fastapi import FastAPI

app = FastAPI()


@app.get("/user/{user}")
def read_user():
    return {"Hello": "World"}

@app.get("/user/me}")
def read_user_me():
   return {"Hello": "World": }
```

### Description

According to https://fastapi.tiangolo.com/tutorial/path-params/?h=+order#order-matters the order of the routes matters. 
Ideally it's a better practice to pay attention to the order, but one can sometimes make mistakes. In my example, the route /user/me will never be called. 

It could maybe be interesting to raise a warning (or an Error?) to inform the user of this. 

### Environment

* OS: Windows
* FastAPI Version 0.60.0

* Python version: 3.8

**Comment by Tyrannas** [comment]
@gfournier

**Comment by ycd** [comment]
I believe this has to be handled on the Starlette side since [this line](https://github.com/encode/starlette/blob/fe961dd22ca69741831d1beb5eac5de0da7d8878/starlette/routing.py#L576-L583) and the lines below, does the routing and matching stuff.

```python
for route in self.routes:
    # Determine if any route matches the incoming scope,
    # and hand over to the matching route if found.
    match, child_scope = route.matches(scope)
    if match == Match.FULL:
        scope.update(child_scope)
        await route.handle(scope, receive, send)
        return
```

**Comment by Tyrannas** [comment]
Totally right ! Thx @ycd , closing the issue :)

**Comment by ycd** [comment]
You're welcome :tada: 

Thanks for closing the issue :+1:

**Comment by tiangolo** [comment]
Thanks for the help here @ycd ! :clap: :bow: 

Thanks for reporting back and closing the issue @Tyrannas  :+1:
